reactor-netty5-http/src/test/java/reactor/netty5/http/server/HttpServerTests.java

```diff
@@ -1913,6 +1917,324 @@ public void channelInboundEvent(ChannelHandlerContext ctx, Object evt) {
+				                .httpRequestDecoder(spec -> spec.h2cMaxContentLength(256)),
+				client -> client.protocol(HttpProtocol.H2C, HttpProtocol.HTTP11),
+				true, true);
+	}
+
+	@Test
+	void testIssue1286ConnectionCloseErrorResponse_H2() throws Exception {
+		Http2SslContextSpec serverCtx = Http2SslContextSpec.forServer(ssc.certificate(), ssc.privateKey());
+		Http2SslContextSpec clientCtx =
+				Http2SslContextSpec.forClient()
+				                   .configure(builder -> builder.trustManager(InsecureTrustManagerFactory.INSTANCE));
+		doTestIssue1286(
+				server -> server.protocol(HttpProtocol.H2).secure(spec -> spec.sslContext(serverCtx)),
+				client -> client.protocol(HttpProtocol.H2).secure(spec -> spec.sslContext(clientCtx)),
+				true, true);
+	}
+
+	@Test
+	void testIssue1286ConnectionCloseErrorResponse_ServerHTTP11AndH2ClientH2() throws Exception {
+		Http2SslContextSpec serverCtx = Http2SslContextSpec.forServer(ssc.certificate(), ssc.privateKey());
+		Http2SslContextSpec clientCtx =
+				Http2SslContextSpec.forClient()
+				                   .configure(builder -> builder.trustManager(InsecureTrustManagerFactory.INSTANCE));
+		doTestIssue1286(
+				server -> server.protocol(HttpProtocol.H2, HttpProtocol.HTTP11).secure(spec -> spec.sslContext(serverCtx)),
+				client -> client.protocol(HttpProtocol.H2).secure(spec -> spec.sslContext(clientCtx)),
+				true, true);
+	}
+
+	@Test
+	void testIssue1286ConnectionCloseErrorResponse_ServerHTTP11AndH2ClientHTTP11AndH2() throws Exception {
+		Http2SslContextSpec serverCtx = Http2SslContextSpec.forServer(ssc.certificate(), ssc.privateKey());
+		Http2SslContextSpec clientCtx =
+				Http2SslContextSpec.forClient()
+				                   .configure(builder -> builder.trustManager(InsecureTrustManagerFactory.INSTANCE));
+		doTestIssue1286(
+				server -> server.protocol(HttpProtocol.H2, HttpProtocol.HTTP11).secure(spec -> spec.sslContext(serverCtx)),
+				client -> client.protocol(HttpProtocol.H2, HttpProtocol.HTTP11).secure(spec -> spec.sslContext(clientCtx)),
+				true, true);
+	}
+
+	private void doTestIssue1286(
+			Function<HttpServer, HttpServer> serverCustomizer,
+			Function<HttpClient, HttpClient> clientCustomizer,
+			boolean connectionClose, boolean throwException) throws Exception {
+		CountDownLatch latch = new CountDownLatch(1);
+		AtomicReference<List<ByteBuf>> byteBufReplay = new AtomicReference<>(new ArrayList<>());
+		AtomicReference<List<Buffer>> bufReplay = new AtomicReference<>(new ArrayList<>());
+		HttpServer server = serverCustomizer.apply(createServer());
+		disposableServer =
+				server.doOnConnection(conn -> conn.addHandlerLast(new ChannelHandlerAdapter() {
+
+				              @Override
+				              public void channelRead(ChannelHandlerContext ctx, Object msg) {
+				                  if (msg instanceof ByteBufHolder byteBuf) {
+				                      byteBufReplay.get().add(byteBuf.content());
+				                  }
```

Comment:
Do we need this? Multipart implementation should use only `Buffer` API or it still uses `ByteBuf` API?

Reply:
applied feedbacks.

The netty-contrib PR is now used from the [httpcontent-used-as-raw-type](https://github.com/pderop/codec-multipart/tree/httpcontent-used-as-raw-type), because the PR is not yet merged.

the CI still reports some errors, I'm now looking into them.